Rdmp.Core/CommandLine/Runners/DleRunner.cs

```diff
@@ -101,8 +103,38 @@ public DleRunner(DleOptions options)
                         execution, databaseConfiguration);
                 }
 
+
                 var exitCode = dataLoadProcess.Run(token);
 
+                if (exitCode is ExitCodeType.Success)
+                {
+                    //Store the date of the last successful load
+                    loadMetadata.LastLoadTime = DateTime.Now;
+                    loadMetadata.SaveToDatabase();
+                    List<IProcessTask> processTasks = loadMetadata.ProcessTasks.Where(ipt => ipt.Path == typeof(RemoteDatabaseAttacher).FullName || ipt.Path == typeof(RemoteTableAttacher).FullName).ToList();
+                    if (processTasks.Count() > 0) //if using a remote attacher, there may be some additional work to do
+                    {
+                        foreach (IEnumerable<Argument> arguments in processTasks.Select(task => task.GetAllArguments()))
+                        {
+                            foreach (var argument in arguments.Where(arg => arg.Name == RemoteAttacherPropertiesValidator("DeltaReadingStartDate") && arg.Value is not null))
+                            {
+                                var scanForwardDate = arguments.Where(a => a.Name == RemoteAttacherPropertiesValidator("DeltaReadingLookForwardDays")).First();
+                                var arg = (ProcessTaskArgument)argument;
+                                arg.Value = DateTime.Parse(argument.Value.ToString()).AddDays(Int32.Parse(scanForwardDate.Value)).ToString();
+                                if (arguments.Where(a => a.Name == RemoteAttacherPropertiesValidator("SetDeltaReadingToLatestSeenDatePostLoad")).First().Value == "True")
```

Comment:
Maybe better as arguments.Single(a => a.Name == RemoteAttacherPropertiesValidator("SetDeltaReadingToLatestSeenDatePostLoad")).Value since there should only be one argument of that name?